doc/page-properties.h

```diff
@@ -0,0 +1,129 @@
+/**********************************************************************
+**
+** Copyright (C) Luxoft Sweden AB 2020
+**
+** This file is part of the documentation of the FaceLift project
+**
+** This work is licensed under the Creative Commons
+** Attribution-ShareAlike 4.0 International License. To view a copy of
+** this license, visit http://creativecommons.org/licenses/by-sa/4.0/
+** or send a letter to Creative Commons, PO Box 1866, Mountain View,
+** CA 94042, USA.
+**
+** SPDX-License-Identifier: CC-BY-SA-4.0
+**
+**********************************************************************/
+
+/**
+\page page-properties Properties
+
+\section properties-intro Introduction
+Properties are the part of the interface definition described in *.qface file. Eventually they will be generated as C++ and QML properties of the interface. For more details about properties in *.qface files see documentation at https://github.com/Pelagicore/qface
+
+In Facelift each property is exposed to the QML engine as Q_PROPERTY with a setter and getter functions, and a change signal. If the property is marked as 'readonly' then no setter is generated.
+
+The facelift::Property class template provides encapsulation for the internal data with a const getter only (except for the service property), which guarantees that the data will not be unnoticibly changed. Each time the setter is called, with a value that differs from the previous one, the change signal is emitted. If the previous value equals to the new value, the change signal will not be emitted.
+
+\section QML
+
+The property is exposed directly to the QML engine by Q_PROPERTY macro and can be used without any additional effort.
+
+\section cpp C++
+
+The property in C++ is generated as one of the templated property containers: Property<T>, ListProperty<T>, ServiceProperty<T> or ModelProperty<T>.
+Additionally ListProperty and QMap<QString, T> partial template specializations add extra convenience functions: void removeAt(int i), void addElement(ElementType element) and int size() const.
```

Comment:
you already mentioned ListProperty. Why "Additionally" ??
you need to mention that QMap<QString,T> is used for map<T> properties defined in qface files.
Why are you getting into API details here ??

Reply:
> you already mentioned ListProperty. Why "Additionally" ??

They are methods that are not present in other specializations.

>  you need to mention that QMap<QString,T> is used for map properties defined in qface files.

OK

>  Why are you getting into API details here ??

Documentation of facelift is very poor. Code is not self explaining and is very complicated, sometimes even spaghetti code. Let's be honest - sometimes not a good quality. Developers should know what are they looking at. 
If we want to keep this project alive, the documentation must be improved. This documentation is not only for the end user but also for developers.